pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/ResultTableResultSet.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+
+/**
+ * ResultSet which contains the ResultTable from the broker response of a sql query.
+ */
+class ResultTableResultSet extends AbstractResultSet {
+  private JsonNode _rowsArray;
+  private JsonNode _columnNamesArray;
+  private JsonNode _columnDataTypesArray;
```

Comment:
Can we replace name and type with schema, and schema can be within metadata field. This way, we can add more metadata as needed.

Reply:
I feel this is okay as is, because:
The purpose of these ResultSets is to get the fields. For a user using the pinot client, once the ResultTable gets wrapped into ResultSet, the ResultSet will be used directly for accessing data. It will not really be converted to other forms further. So this is just internal representation.

For metadata, this is easy to change later. That will only be necessary when more APIs are added to the ResultSet. The current set of APIs are answered fine with this structure.

Lmk if you think it's okay.